File: QViz.NET/Periscope.QViz/JSON/UserProfile.cs
using System;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for User Profiles
	/// </summary>
	public class UserProfile
	{
		/// <summary>
		/// Identifier of the User Profile
		/// </summary>
		public string userProfilesId;
		
		/// <summary>
		/// Identifier of the User with the Profile
		/// </summary>
		public string userId;
		
		/// <summary>
		/// Identifier of the Project associated with the User Profile
		/// </summary>
		public string projectId;
		
		/// <summary>
		/// Identifier of the Role associated with the User Profile
		/// </summary>
		public string roleId;
		
		/// <summary>
		/// Is this User Profile Active?
		/// </summary>
		public bool isActive;
		
		/// <summary>
		/// Date and Time on which this Model was Created
		/// </summary>
		public DateTime createdOn;
		
		/// <summary>
		/// Date and Time on which this Model was last Updated
		/// </summary>
		public DateTime updatedOn;
		
		/// <summary>
		/// User Object associated with the Profile
		/// </summary>
		public User user;
		
		/// <summary>
		/// Role Object associated with the User Profile
		/// </summary>
		public Role role;
		
		/// <summary>
		/// Project Object associated with the User Profile
		/// </summary>
		public Project project;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public UserProfile()
		{
			this.userProfilesId = "";
			this.userId = "";
			this.projectId = "";
			this.roleId = "";
			this.isActive = false;
			this.createdOn = DateTime.Now;
			this.updatedOn = DateTime.Now;
			this.user = new User();
			this.role = new Role();
			this.project = new Project();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/UserProfile.java
package com.periscope.qviz.json;

import java.util.Date;

/**
 * JSON Model for User Profiles
 */
public class UserProfile {

	/**
	 * Identifier of the User Profile
	 */
	public String userProfilesId;

	/**
	 * Identifier of the User with the Profile
	 */
	public String userId;

	/**
	 * Identifier of the Project associated with the User Profile
	 */
	public String projectId;

	/**
	 * Identifier of the Role associated with the User Profile
	 */
	public String roleId;

	/**
	 * Is this User Profile Active?
	 */
	public Boolean isActive;

	/**
	 * Is this User has access to Sensitive data?
	 */
	public Boolean allowSensetiveData;

	/**
	 * Date and Time on which this Model was Created
	 */
	public Date createdOn;

	/**
	 * Date and Time on which this Model was last Updated
	 */
	public Date updatedOn;

	/**
	 * User Object associated with the Profile
	 */
	public User user;

	/**
	 * Role Object associated with the User Profile
	 */
	public Role role;

	/**
	 * Project Object associated with the User Profile
	 */
	public Project project;

	/**
	 * Default Constructor
	 */
	public UserProfile() {
		this.userProfilesId = "";
		this.userId = "";
		this.projectId = "";
		this.roleId = "";
		this.isActive = false;
		this.allowSensetiveData = false;
		this.createdOn = null;
		this.updatedOn = null;
		this.user = new User();
		this.role = new Role();
		this.project = new Project();
	}

}
